app/content/views/user.py

```diff
@@ -273,14 +273,41 @@ def get_user_detail_strikes(self, request, *args, **kwargs):
         )
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+    # @action(detail=False, methods=["get"], url_path="me/events")
+    # def get_user_events(self, request, *args, **kwargs):
+    #     registrations = request.user.registrations.all()
+    #     events = [
+    #         registration.event
+    #         for registration in registrations
+    #         if not registration.event.expired
+    #     ]
+    #     return self.paginate_response(
+    #         data=events, serializer=EventListSerializer, context={"request": request}
+    #     )
+
     @action(detail=False, methods=["get"], url_path="me/events")
     def get_user_events(self, request, *args, **kwargs):
         registrations = request.user.registrations.all()
+
+        # Apply the filter
+        event_has_ended = self.request.query_params.get("expired") == "true"
+
+        # expired_field = self.request_params.get("expired")
+        # filter_expired = CaseInsensitiveBooleanQueryParam(expired_field) / teste for None
+        #
+        #  events = [
+        #    registration.event
+        #    for registration in registrations
+        #    if registration.event.expired == filter_expired
+        #  ]
+
+        # todo tom query -> = false
         events = [
             registration.event
             for registration in registrations
-            if not registration.event.expired
+            if registration.event.expired == event_has_ended
         ]
```

Comment:
Bra! Men dette returnerer nå bare events hvis filteret er på, dere må legge til en else, som sender med events som ikke er expired hvis filteret ikke er på.

slik:

if event_has_ended:
  events = [
            registration.event
            for registration in registrations
            if registration.event.expired == event_has_ended
        ]
else:
  events = [
            registration.event
            for registration in registrations
            if not registration.event.expired
        ]

Reply:
Mener dette ikke er nødvendig. Event_has_ended er en boolean som beskriver om filteret skal filtrere for:

-  "expired" events (når query paramsa er "true" ==> event_has_ended = true) 
- "unexpired" events (når query paramsa er alt annet enn "true" ==> event_has_ended = false)

Så basicaly, når event_has_ended = true, skal kun expired events legges til. Når event_has_ended = false, skal kommende events legges til.

Som et eksempel, si query params er "false". Dermed er event_has_ended = false. I for-loopen skal vi velge events bare hvis expired = false (altså at eventet er kommende). Kun kommende events legges dermed til i event-listen.

For den motsatte casen, la event_has_ended = true. Her vil for-loopen kun velge events hvis expired = true. Alle expired events legges dermed til i event-listen.

Reply:
Det stemmer! Kom for å endre det nå, bra du er oppmerksom:)